examples/chat/proto/chat.proto

```diff
@@ -64,5 +49,5 @@ service Chat {
   rpc Subscribe(SubscribeRequest) returns (stream Message);
 
   // Send a message to a chat room.
```

Comment:
I agree with the philosophical point above about what a client device cares about, but the model is not great from a developer point of view. The developer now has to think about every API in terms of this split view. Validating consistency between data and metadata is more difficult as these are not tightly coupled. This is further complicated by the metadata being defined per connection and the data per message.

Reply:
I have added some more comments here.

Conceptually this is similar to how in an API you may have methods that return data based on the provided authentication credentials, for instance all email messages for the authenticated user, etc. Though here the relationship is complicated by the fact that the label of the request is used to select the chat room to access. Perhaps we should go back to using some other identifier for chat rooms instead of the label itself, but then it would not be possible to route based on it because the room identifier would be protected by the label and not accessible by the router node. I think this may be solvable by an additional layer of routing nodes, but I am not sure that's the best way to go, so I'd leave that experiment and discussion for a future PR.

I also added a new method `GetAllMessagesCount`, which is supposed to return the number of messages from all chats that the user has access to. It does not seem practical (possible?) for it to be implemented in the current system, so I left it unimplemented, but I think it is a good discussion point to consider for future label-related discussions.

Reply:
Update: I removed `GetAllMessagesCount` again, I'll leave that for another time, since currently it does not really add much value.